addons/sale/models/sale_order_line.py

```diff
@@ -485,6 +485,10 @@ def _compute_product_uom_qty(self):
 
     @api.depends('product_id')
     def _compute_product_uom_id(self):
+        if (old := self._origin) and old.product_id and self.product_id and \
+                self.product_id != old.product_id and \
+                self.product_template_id == old.product_template_id:
+            return
```

Comment:
```suggestion
        if (
            (old := self._origin)
            and old.product_id
            and self.product_id
            and self.product_id != old.product_id
            and self.product_template_id == old.product_template_id
        ):
            return
```

Also, will it work if I change from product A to product B (different templates) then product A again ? Won't it keep the uom of product B ?

Reply:
Indeed, changing the template will reset the UOM. Do we want to remember the previous choices ?